lib/actions/ExtensionAction.js

```diff
@@ -475,6 +518,223 @@ class ExtensionAction {
     if (toBeAttached.length) await this.attachExtensions({ extensions: toBeAttached })
   }
 
+  /**
+   * @param {Object} args
+   * @param {String} [args.extension] Target extension folder name
+   * @param {Object} options
+   * @param {Number} [options.preprocessTimeout] Number of milliseconds to wait for preprocessing to complete
+   * @param {Boolean} [options.force] Forces creation of version without asking user
+   * @returns {Promise}
+   */
+  async uploadExtension (args = {}, options = {}) {
```

Comment:
This function is way too long with almost 200 lines of code.
It should be splitted in multiple functions.

I think for the better maintainability its also better to move this code to another file and just call it here

Reply:
I split the function into blocks. Should be better now. Please take a look.